### PR TITLE
Implement underline,strikethrough styling for Label and MarkupLabel

### PR DESCRIPTION
**Pull request by udiboy1209**
ref #3903

Added two init boolean arguments 'underline' and 'strikethrough' in kivy.uix.label.Label and two options of same name in kivy.core.text.LabelBase 

Uses the SDL routine TTF_SetFontStyle()

Added tags to MarkupLabel for the same.

### CODE REVIEW

**Comment by udiboy1209** [comment]
@dessant please review this.

**Comment by dessant** [comment]
Looks good @udiboy1209, great job. I'll test it today.
New api endpoints need a versionadded tag and a bit of docstrings, you could use this as a template for those two props: https://github.com/udiboy1209/kivy/blob/styling/kivy/uix/label.py#L619.
The version will be `2.0.0`.

**Comment by akshayaurora** [comment]
We need to support all providers not just sadly so support for this needs to be added to text_pygame, text_pil,...

**Comment by udiboy1209** [comment]
@akshayaurora @dessant should I mention that it only works with SDL2 in the docstring? the issue did not mention support for pygame,pil... only SDL2

**Comment by dessant** [comment]
@udiboy1209 yes, it should be mentioned, like this: https://github.com/kivy/kivy/blob/4e429deb8864e3450cb71cd670cd00d1516d2799/kivy/core/window/__init__.py#L837.

**Comment by matham** [comment]
Why is the version 2.0.0? We're at 1.9.2.

**Comment by dessant** [comment]
@matham, we may need to update the repo version too.

File: kivy/uix/label.py [hunk 1]
```diff
@@ -497,6 +498,33 @@ def on_ref_press(self, ref):
     to False.
     '''
 
+    underline = BooleanProperty(False)
+    '''Adds an underline to the text
+
+    .. note::
+        This feature requires a SDL2 window provider and is currently only
+        supported on desktop platforms.
```
Comment (dessant) [inline kivy/uix/label.py]: It is supported everywhere.

File: kivy/uix/label.py [hunk 2]
```diff
@@ -497,6 +498,33 @@ def on_ref_press(self, ref):
     to False.
     '''
 
+    underline = BooleanProperty(False)
+    '''Adds an underline to the text
```
Comment (dessant) [inline kivy/uix/label.py]: Missing period. :laughing:

File: kivy/uix/label.py [hunk 3]
```diff
@@ -497,6 +498,33 @@ def on_ref_press(self, ref):
...
+        This feature requires a SDL2 window provider and is currently only
+        supported on desktop platforms.
+
+    .. versionadded:: 2.0.0
+
+    :attr:`strikethrough` is a :class:`~kivy.properties.BooleanProperty` and defaults
+    to False.
+    '''
+
+
```
Comment (dessant) [inline kivy/uix/label.py]: Please delete this line.

**Comment by dessant** [comment]
The code works flawlessly, great work. :+1: 

capture112

**Comment by dessant** [comment]
Awesome, thanks a lot for your contribution. :wink:

**Comment by udiboy1209** [comment]
@dessant you guys check every little detail, even in the docstrings :stuck_out_tongue: ! Happy to contribute! :smile:

**Pull request merged**